docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,271 @@
+
+#### command service configuration
+
+Example command service configuration is provided below.
+
+``` toml
+[MessageQueue]
+    [InternalMessageQueue]
+    Protocol = "redis"
+    Host = "localhost"
+    Port = 6379
+    Type = "redis"
+    AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
+    SecretName = "redisdb"
+    [ExternalMessageQueue]
+    Protocol = "redis"
+    Host = "localhost"
+    Port = 6378
+    Type = "redis"
+    AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
+    SecretName = "redisdb"
+ExternalSubscribeRequestTopic = “edgex/command/request/#”
+InternalPublishRequestTopicPrefix = "edgex/command/request/"  # <device-service>/<device-name>/<command-name>/<method> will be added to this publish topic prefix
+InternalSubscribeResponseTopic = “edgex/command/response/#”
+ExternalPublishResponseTopicPrefix = “edgex/command/response/"  # /<device-name>/<command-name>/<method> will be added to this publish topic prefix
+```
+
+#### device service configuration
+
+Example device service configuration is provided below.
+
+``` toml
+[MessageQueue]
+Protocol = "redis"
+Host = "localhost"
+Port = 6379
+Type = "redis"
+AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
+SecretName = "redisdb"
+SubscribeRequestTopic = "edgex/command/request/#"
+PublishResponseTopicPrefix = “edgex/command/response/"  # <device-service>/<device-name>/<command-name>/<method> will be added to this publish topic prefix
+```
+
+## Questions
+
+- Do we need separate topics for all the devices or would one on the device service suffice?
+    
+    - Ans:  we have defined the deviceName in the payload, so one topic should be sufficient for Device Service- edgex/command/request/<device-name>...
+
+- Would clients (non EdgeX services and applications) want to get a list of available commands via message (instead of calling REST)?
+
+    - Ans:  this is a valid question and could be provided via later additions to the command service (or other service like metadata) in the future.  It does not have to be tackled immediately.  
+
+- Dynamic configuration of the message subscription is not a user friendly operation today (requiring configuration changes).
+
+    - Ans:  We might want to think about creating additional APIs for Adding/Updating/Deleting/Query the external subscription (and store them to the RedisDB).
+
+- Is it acceptable for more than one response to be published by the device service on the same correlation ID? Eg, send back "Acknowledged", then "Scheduled", then "Starting", then "Done" statuses?
+    
+    - Ans:  No, the correlation id has a life span to/from the initial requester to the response back to the requester.
+
+- Would it make sense to echo the command name into the response, as a reality check?
+    - Ans: solved via topic naming
```

Comment:
IMO, not needed as we don't do this in the HTTP response. As commened above the repose topic doesn't need the extra path info. The request ID or correlation ID is all that is needed to match the response to the request. No need to make it more complex.

Reply:
Added Lenny's comment.